Format removes empty tuples from Impl but not from enum

I had something like

```rust
use std::convert::Into;

/// Known paths
pub enum Paths<'a> {
    Batch(),
    Buckets(),
    Bucket(&'a str)
}

impl<'a> Into<String> for Paths<'a> {
    fn into(self) -> String {
        match self {
            Paths::Batch() => format!("/batch"),
            Paths::Buckets() => format!("/buckets"),
            Paths::Bucket(id) => format!("/buckets/{id}", id = id),
        }
    }
}
```

After running format, it removed the empty tuples from the implementation match but not from the the enum definition. Leading to some errors. You can see the format result and the compiler output. 

```rust
use std::convert::Into;

/// Known paths
pub enum Paths<'a> {
    Batch(),
    Buckets(),
    Bucket(&'a str)
}

impl<'a> Into<String> for Paths<'a> {
    fn into(self) -> String {
        match self {
            Paths::Batch => format!("/batch"),
            Paths::Buckets => format!("/buckets"),
            Paths::Bucket(id) => format!("/buckets/{id}", id = id),
        }
    }
}
```
```
gsurita-30820:kinto-http.rs gsurita$ RUST_TEST_THREADS=1 RUST_BACKTRACE=1 cargo test 
   Compiling kinto_http v0.1.0 (file:///Users/gsurita/kinto-http.rs)
error[E0532]: expected unit struct/variant or constant, found tuple variant `Paths::Batch`
  --> src/paths.rs:21:13
   |
21 |             Paths::Batch => format!("/batch"),
   |             ^^^^^^^^^^^^ not a unit struct/variant or constant

error[E0532]: expected unit struct/variant or constant, found tuple variant `Paths::Buckets`
  --> src/paths.rs:22:13
   |
22 |             Paths::Buckets => format!("/buckets"),
   |             ^^^^^^^^^^^^^^ not a unit struct/variant or constant
```

--- a/src/patterns.rs
+++ b/src/patterns.rs
@@ -260,55 +260,54 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
     }
 
     if pat_vec.is_empty() {
-        path_str
+        return Some(format!("{}()", try_opt!(path_str)));
+    }
+    // add comma if `(x,)`
+    let add_comma = path_str.is_none() && pat_vec.len() == 1 && dotdot_pos.is_none();
+
+    let path_len = path_str.as_ref().map(|p| p.len()).unwrap_or(0);
+    // 2 = "()".len(), 3 = "(,)".len()
+    let nested_shape = try_opt!(shape.sub_width(path_len + if add_comma { 3 } else { 2 }));
+    // 1 = "(".len()
+    let nested_shape = nested_shape.visual_indent(path_len + 1);
+    let mut items: Vec<_> = itemize_list(context.codemap,
+                                         pat_vec.iter(),
+                                         if add_comma { ",)" } else { ")" },
+                                         |item| item.span().lo,
+                                         |item| item.span().hi,
+                                         |item| item.rewrite(context, nested_shape),
+                                         context.codemap.span_after(span, "("),
+                                         span.hi - BytePos(1))
+            .collect();
+
+    // Condense wildcard string suffix into a single ..
+    let wildcard_suffix_len = count_wildcard_suffix_len(&items);
+
+    let list = if context.config.condense_wildcard_suffices && wildcard_suffix_len >= 2 {
+        let new_item_count = 1 + pats.len() - wildcard_suffix_len;
+        items[new_item_count - 1].item = Some("..".to_owned());
+
+        let da_iter = items.into_iter().take(new_item_count);
+        try_opt!(format_item_list(da_iter, nested_shape, context.config))
     } else {
-        // add comma if `(x,)`
-        let add_comma = path_str.is_none() && pat_vec.len() == 1 && dotdot_pos.is_none();
+        try_opt!(format_item_list(items.into_iter(), nested_shape, context.config))
+    };
 
-        let path_len = path_str.as_ref().map(|p| p.len()).unwrap_or(0);
-        // 2 = "()".len(), 3 = "(,)".len()
-        let nested_shape = try_opt!(shape.sub_width(path_len + if add_comma { 3 } else { 2 }));
-        // 1 = "(".len()
-        let nested_shape = nested_shape.visual_indent(path_len + 1);
-        let mut items: Vec<_> = itemize_list(context.codemap,
-                                             pat_vec.iter(),
-                                             if add_comma { ",)" } else { ")" },
-                                             |item| item.span().lo,
-                                             |item| item.span().hi,
-                                             |item| item.rewrite(context, nested_shape),
-                                             context.codemap.span_after(span, "("),
-                                             span.hi - BytePos(1))
-                .collect();
-
-        // Condense wildcard string suffix into a single ..
-        let wildcard_suffix_len = count_wildcard_suffix_len(&items);
-
-        let list = if context.config.condense_wildcard_suffices && wildcard_suffix_len >= 2 {
-            let new_item_count = 1 + pats.len() - wildcard_suffix_len;
-            items[new_item_count - 1].item = Some("..".to_owned());
-
-            let da_iter = items.into_iter().take(new_item_count);
-            try_opt!(format_item_list(da_iter, nested_shape, context.config))
-        } else {
-            try_opt!(format_item_list(items.into_iter(), nested_shape, context.config))
-        };
-
-        match path_str {
-            Some(path_str) => {
-                Some(if context.config.spaces_within_parens {
-                         format!("{}( {} )", path_str, list)
-                     } else {
-                         format!("{}({})", path_str, list)
-                     })
-            }
-            None => {
-                let comma = if add_comma { "," } else { "" };
-                Some(if context.config.spaces_within_parens {
-                         format!("( {}{} )", list, comma)
-                     } else {
-                         format!("({}{})", list, comma)
-                     })
-            }
+    match path_str {
+        Some(path_str) => {
+            Some(if context.config.spaces_within_parens {
+                     format!("{}( {} )", path_str, list)
+                 } else {
+                     format!("{}({})", path_str, list)
+                 })
+        }
+        None => {
+            let comma = if add_comma { "," } else { "" };
+            Some(if context.config.spaces_within_parens {
+                     format!("( {}{} )", list, comma)
+                 } else {
+                     format!("({}{})", list, comma)
+                 })
         }
     }
 }

--- a/tests/target/empty-tuple-no-conversion-to-unit-struct.rs
+++ b/tests/target/empty-tuple-no-conversion-to-unit-struct.rs
@@ -1,0 +1,13 @@
+enum TestEnum {
+    Arm1(),
+    Arm2,
+}
+
+
+fn foo() {
+    let test = TestEnum::Arm1;
+    match test {
+        TestEnum::Arm1() => {}
+        TestEnum::Arm2 => {}
+    }
+}

